libs/libcommon/include/common/Types.h

```diff
@@ -53,3 +53,61 @@ struct is_arithmetic
 template <typename T>
 inline constexpr bool is_arithmetic_v = is_arithmetic<T>::value;
 
+struct UInt8NoAlias
+{
+    uint8_t value;
+    operator uint8_t & () { return value; }
+    operator const uint8_t & () const { return value; }
+    UInt8NoAlias() {}
+    template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+    UInt8NoAlias(const T & o) : value(o) {}
+    template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+    UInt8NoAlias operator=(const T & o) { return value = o; }
+
+    bool inline operator== (const UInt8NoAlias rhs) const { return value == rhs.value; }
+    bool inline operator!= (const UInt8NoAlias rhs) const { return value != rhs.value; }
+    bool inline operator<  (const UInt8NoAlias rhs) const { return value < rhs.value; }
+    bool inline operator<= (const UInt8NoAlias rhs) const { return value <= rhs.value; }
+    bool inline operator>  (const UInt8NoAlias rhs) const { return value > rhs.value; }
+    bool inline operator>= (const UInt8NoAlias rhs) const { return value >= rhs.value; }
+
+    template <typename T> bool inline operator== (const T rhs) const { return value == rhs; }
+    template <typename T> bool inline operator!= (const T rhs) const { return value != rhs; }
+    template <typename T> bool inline operator>= (const T rhs) const { return value >= rhs; }
+    template <typename T> bool inline operator>  (const T rhs) const { return value >  rhs; }
+    template <typename T> bool inline operator<= (const T rhs) const { return value <= rhs; }
+    template <typename T> bool inline operator<  (const T rhs) const { return value <  rhs; }
+};
+
+bool max(UInt8NoAlias a, UInt8 b);
+bool max(UInt8 a, UInt8NoAlias b);
```

Comment:
This can be dangerous if this function with implicit cast will be preferred over std.